### ISSUE TITLE
validate function in vatvalidation calls checkVatNumber a lot

### ISSUE THREAD

**Issue by sanderjongsma** [magento/magento2#17658]
I discovered this issue because I use the following module: [Dutchento_Vatfallback](https://github.com/Dutchento/m2-vatfallback). In here I configured Vatlayer so vatnumbers within Magento are validated through their API. No I saw the calls to their API where extreme high, so I went debugging first in the module and after that in Magento itself. After a while I came across the `validate` function in the file `magento/module-quote/Observer/Frontend/Quote/Address/VatValidator.php`. 

```
public function validate(\Magento\Quote\Model\Quote\Address $quoteAddress, $store)
    {
        $customerCountryCode = $quoteAddress->getCountryId();
        $customerVatNumber = $quoteAddress->getVatId();

        $merchantCountryCode = $this->customerVat->getMerchantCountryCode();
        $merchantVatNumber = $this->customerVat->getMerchantVatNumber();

        $validationResult = null;
        if ($this->customerAddress->hasValidateOnEachTransaction(
            $store
        ) ||
            $customerCountryCode != $quoteAddress->getValidatedCountryCode() ||
            $customerVatNumber != $quoteAddress->getValidatedVatNumber()
        ) {
            // Send request to gateway
            $validationResult = $this->customerVat->checkVatNumber(
                $customerCountryCode,
                $customerVatNumber,
                $merchantVatNumber !== '' ? $merchantCountryCode : '',
                $merchantVatNumber
            );

            // Store validation results in corresponding quote address
            $quoteAddress->setVatIsValid((int)$validationResult->getIsValid());
            $quoteAddress->setVatRequestId($validationResult->getRequestIdentifier());
            $quoteAddress->setVatRequestDate($validationResult->getRequestDate());
            $quoteAddress->setVatRequestSuccess($validationResult->getRequestSuccess());
            $quoteAddress->setValidatedVatNumber($customerVatNumber);
            $quoteAddress->setValidatedCountryCode($customerCountryCode);
            $quoteAddress->save();
        } else {
            // Restore validation results from corresponding quote address
            $validationResult = new \Magento\Framework\DataObject(
                [
                    'is_valid' => (int)$quoteAddress->getVatIsValid(),
                    'request_identifier' => (string)$quoteAddress->getVatRequestId(),
                    'request_date' => (string)$quoteAddress->getVatRequestDate(),
                    'request_success' => (bool)$quoteAddress->getVatRequestSuccess(),
                ]
            );
        }

        return $validationResult;
    }
```

Before calling `$this->customerVat->checkVatNumber` this function checks:

```
$customerCountryCode != $quoteAddress->getValidatedCountryCode() ||
$customerVatNumber != $quoteAddress->getValidatedVatNumber()
```

I expected validated addresses would be stored in de db because of this check, but after digging around, `$quoteAddress->getValidatedCountryCode()` and `$quoteAddress->getValidatedVatNumber()` are doing nothing at all, so it is expected they are quote_address attributes in the db. 

But in the file `magento/module-quote/Setup/InstallData.php` only the following attributes are installed:

```
$attributes = [
            'vat_id' => ['type' => Table::TYPE_TEXT],
            'vat_is_valid' => ['type' => Table::TYPE_SMALLINT],
            'vat_request_id' => ['type' => Table::TYPE_TEXT],
            'vat_request_date' => ['type' => Table::TYPE_TEXT],
            'vat_request_success' => ['type' => Table::TYPE_SMALLINT],
        ];
```

but `validated_country_code` and `validated_vat_number` are missing and nowhere to be found.

When I installed the missing attributes with a custom InstallData script everything works as expected.

### Preconditions
1. Magento 2.2.4
2. Php 7.1
3. Mysql 5.7
4. Docker

### Steps to reproduce
To reproduce this I recommend setting up vat validation with help from the [docs](https://docs.magento.com/m2/ce/user_guide/tax/vat-validation-configure.html) and enable breakpoints with xdebug within the `validate` function and check how often `$this->customerVat->checkVatNumber` is called. `validate` is triggered by an Observer: `magento/module-quote/Observer/Frontend/Quote/Address/CollectTotalsObserver.php` so it is called a lot. Try adding a product to the cart for example.

### Expected result
The expected result is the check inside the `validate` function: 

```
$customerCountryCode != $quoteAddress->getValidatedCountryCode() ||
$customerVatNumber != $quoteAddress->getValidatedVatNumber()
```

will actually do something.

### Actual result
The checks are now doing nothing because the attributes in quote_address don't exist. When I install the missing attributes: `validated_country_code` and `validated_vat_number` with a custom InstallData script everything works as expected.

**Comment by magento-engcom-team** [comment]
Hi @sanderjongsma. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@sanderjongsma do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by engcom-backlog-nickolas** [comment]
Hello @sanderjongsma, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @sanderjongsma. Thank you for your report. 
The issue has been fixed in magento/magento2#19265 by @ErikPel in 2.3-develop branch
Related commit(s):
 - [e27f09f987f289b0a2d3f1e0552f67c63e2e302d](https://github.com/magento/magento2/commit/e27f09f987f289b0a2d3f1e0552f67c63e2e302d)
 - [7f15bd86241bace94cdc77b5487bc7f593482a69](https://github.com/magento/magento2/commit/7f15bd86241bace94cdc77b5487bc7f593482a69)

The fix will be available with the upcoming 2.3.2 release.
